### DISCUSSION TITLE
Reducing overall CPU load of Frigate

### DISCUSSION THREAD

**Discussion by alinbalutoiu** [blakeblackshear/frigate#5757] [Ask A Question] [answered]
Hello there! I am wondering if it is possible to reduce the CPU load for Frigate.
In my use case, I've got Frigate with birdseye running with continuous mode turned on.

I've had a look at the system page, and I'm wondering about the following:
- the `detect` process for a camera seems pretty useless in my case, is there a reason why that shouldn't be disabled? (i.e. if you don't run Birdseye with `motion` or `objects`)
- what is the responsibility of the `Capture` process?
- what is the responsibility of the `ffmpeg` process?

I've enabled the [NVIDIA hardware acceleration](https://docs.frigate.video/configuration/hardware_acceleration#nvidia-gpu) and I was expecting to see a significant load decrease in the CPU usage, but that seems to be quite small (CPU usage decreased from 60% on average to 50%).

**Comment by NickM-27** [comment, marked as answer]
> I've had a look at the system page, and I'm wondering about the following:
> 
> * the `detect` process for a camera seems pretty useless in my case, is there a reason why that shouldn't be disabled? (i.e. if you don't run Birdseye with `motion` or `objects`)

The detect process always runs even when object detection is disabled. This can not be turned off and is responsible for motion and objects modes. 

> * what is the responsibility of the `Capture` process?

The capture process is the process which manages the ffmpeg process, ensures it is restarted if it crashes, makes sure complete frames continue to be sent from the process, etc.

> * what is the responsibility of the `ffmpeg` process?

ffmpeg is used to take a cameras stream which is encoded in (typically) h.264 or h.265 and decode those compressed streams into individual frames which is what feeds birdseye, detect, snapshots, etc.

**Reply by NickM-27** [reply]
I'd suggest reading https://github.com/blakeblackshear/frigate/issues/3860
